### DISCUSSION TITLE
[HELP] Cómo puedo validar un correo (el formato)?

### DISCUSSION THREAD

**Discussion by NicolasPLG** [Cotalker/documentation#210] [2. Questions and Answers] [answered]
Hola

Creo que al usar este redex: /^[-\w.%+]{1,64}@(?:[A-Z0-9-]{1,63}\.){1,125}[A-Z]{2,63}$/i me podría servir?, lo busqué por la web.

Ahora la pregunta cómo lo coloco?, ya que revisé en las questions y no encontré una similar (de las 30k aprox).
Creo que debería ser un exec validate, pero como colocaría la expresión?

Gracias :>

**Comment by guillevalin** [comment, marked as answer]
@NicolasPLG tienes que usar un questionExec de tipo "validate", cuya documentación se encuentra en la siguiente página:

https://doc.cotalker.com/docs/documentation/automation/question_exec

Justamente ahí existe un ejemplo que valida un regex (Ejemplo 3). Sólo deberías cambiar el regex de validación.

**Reply by NicolasPLG** [reply]
hola @guillevalin lo que entendería es cambiar esto: if (!email.match(/@/))  por esto if (!email.match(/^[-\w.%+]{1,64}@(?:[A-Z0-9-]{1,63}.){1,125}[A-Z]{2,63}$/)) y debería funcionar correctamente? :>?

**Reply by guillevalin** [reply]
Si... solo cambiar el regex @NicolasPLG

**Reply by NicolasPLG** [reply]
funciona

gracias @guillevalin 

Dejaré el código de la question por la respuesta:

````
{ 
    "_id" : ObjectId("607d913fc21cba000898b367"), 
    "command" : {
        "commands" : [

        ], 
        "resetIdentifiers" : [

        ], 
        "isCommanded" : null, 
        "values" : [

        ]
    }, 
    "display" : [

    ], 
    "code" : [

    ], 
    "isActive" : true, 
    "isSystemModel" : false, 
    "contentType" : "application/vnd.cotalker.survey+textinput", 
    "help" : null, 
    "identifier" : "correo_electronico_contacto", 
    "isReadOnly" : false, 
    "max" : NumberInt(5000), 
    "min" : NumberInt(1), 
    "required" : true, 
    "subtype" : "short_text", 
    "company" : ObjectId("606f1419a33b5d016fb3d1d1"), 
    "modifiedAt" : ISODate("2021-05-26T17:02:31.449+0000"), 
    "__v" : NumberInt(0), 
    "exec" : {
        "validate" : {
            "src" : "function run() {\n  const email = context['responses#self'][0] ; // gets the value from the input\n  \n  if (!email.match(/^[-\\w.%+]{1,64}@(?:[A-Z0-9-]{1,63}.){1,125}[A-Z]{2,63}$/i)) \n    return [\n      { \n        cmd: 'RESULT', \n        result: false, \n        value: 'No es un formato de correo' \n      }\n    ];\n\n  return [{ cmd: 'RESULT', result: true }];\n}", 
            "context" : "responses#self"
        }
    }
}

```
